python/tests/stress/arcticdb/test_mem_leaks.py

```diff
@@ -0,0 +1,176 @@
+
+    mem_growth_each_iter = list()
+
+    for n in range(number_iterrations):
+        p_iter_mem_start : np.int64  = p.memory_info().rss
+        
+        print("Starting watched code ...........")
+        process_func()
+        lets_collect_some_garbage()
+
+        p_iter_mem_end : np.int64  = p.memory_info().rss
+        process_growth : np.int64 = p.memory_info().rss - process_initial_memory
+        mem_growth_each_iter.append(process_growth)
+        process__avg_growth = process_growth / (n + 1)
+        p_iter_mem_notcleaned : np.int64  = p_iter_mem_end - p_iter_mem_start
+        mem_avail : np.int64 = psutil.virtual_memory().available
+        mem_per : np.int64 = psutil.virtual_memory().percent
+        print (f"Iter No[{n}]    : Process did added (or if - number means cleaned) {nice_bytes_str(p_iter_mem_notcleaned)}.  Avail memory: {nice_bytes_str(mem_avail)} Used Mem: {mem_per}%")
+        print (f"  Overall stats : Process growth since start {nice_bytes_str(process_growth)} AVG growth per iter {nice_bytes_str(process__avg_growth)}")
+        print (f"  Minimum growth so far: {nice_bytes_str(min(mem_growth_each_iter))}")
+        print (f"  Maximum growth so far: {nice_bytes_str(max(mem_growth_each_iter))}")
+        print (f"  Number of times there was 50% drop in memory: {count_drops(mem_growth_each_iter, 0.5)}")
+
+        assert max_total_mem_lost_treshold >= process_growth , f"Memory of the process grew more than defined threshold: {nice_bytes_str(process_growth)} (specified: {nice_bytes_str(max_total_mem_lost_treshold)} )"
+        assert max_machine_memory_percentage >= mem_per , f"Machine utilized more memory than specified treshold :{mem_per}% (specified {max_machine_memory_percentage}%)"
+        
+    print ("The process assesment finished within expectations. Total consumed additionaml mem is bellow threshold: ", process_growth)
+    print ("Returning this number as result.")
+    return process_growth, count_drops(mem_growth_each_iter, 0.5)
+
+def grow_exp(df_to_grow : pd.DataFrame, num_times_xx2:int):
+    """
+        Quickly inflates the dataframe by doubling its size for each iterration
+        Thus the final size will be 'num_times_xx2' power of two
+    """
+    for n in range(1,num_times_xx2):
+        df_prev = df_to_grow.copy(deep=True)
+        df_to_grow = pd.concat([df_to_grow, df_prev])
+    return df_to_grow
+
+@pytest.mark.xfail("ArcticDBPR#1998", reason = "String memory leaks while read")
+def test_mem_leak_read_all_arctic_lib(arctic_library_lmdb):
+    lib : adb.Library = arctic_library_lmdb
+
+    df = get_sample_dataframe(size=1000000)
+    df = grow_exp(df, 5)
+
+    symbol = "test"
+    lib.write(symbol, df)
+
+    def proc_to_examine():
+        data = lib.read(symbol).data
+        del data
+
+    df = lib.read(symbol).data
+    print(df)
+    del df
+
+    check_process_memory_leaks(proc_to_examine, 30, 4808662528, 80.0)
```

Comment:
I would make 4808662528 into a variable and break down how it was calculated so it is easier to see the value that is tests.
Same goes for 808662528 below.